dissect/cstruct/expression.py

```diff
@@ -18,7 +18,7 @@ def __init__(self, expression: str):
         self.pos = 0
         self.tokens = []
 
-    def equal(self, token: str, expected: Union[str, str[str]]) -> bool:
+    def equal(self, token: str, expected: Union[str, set[str]]) -> bool:
```

Comment:
```suggestion
    def equal(self, token: str, expected: str | set[str]) -> bool:
```
May as well use the new syntax here too.